.changeset/modern-ghosts-sin.md

```diff
@@ -0,0 +1,41 @@
+---
+"@logto/tunnel": minor
+---
+
+add deploy command and env support
+
+#### Add new `deploy` command to deploy your local custom UI assets to your Logto Cloud tenant
+
+1. Create a machine-to-machine app in your Logto tenant
+2. Run the following command
+
+```bash
+npx @logto/tunnel deploy --auth <your-m2m-app-id>:<your-m2m-app-secret> --endpoint https://<tenant-id>.logto.app --experience-path /path/to/your/custom/ui
+```
+
+If you are using custom domain for your Logto endpoint, an additional option `--management-api-resource` (or `--resource`) must be specified. 
+
+```bash
+npx @logto/tunnel deploy --auth <your-m2m-app-id>:<your-m2m-app-secret> --endpoint https://your.custom.domain --experience-path /path/to/your/custom/ui --management-api-resource https://<tenant-id>.logto.app/api
+```
+
+Note:
+1. This option can be omitted when using the default Logto domain, since the CLI can infer the value automatically.
+2. Using either custom domain or default Logto domain as the endpoint will yield identical results.
```

Comment:
Maybe we can mention that `https://<tenant-id>.logto.app/api` is used internally within Logto's system, so if we're using custom domain (for end users), we need to provide this resource identifier.
Because I will confused about why I need to add this resource and why it will be infered automatically when using the default Logto domain.

Reply:
Because the resource is just the default domain endpoint appended with `/api`. So I think it would not be difficult to understand the relations. Right?